consensus/polybft/state_store_state_sync.go

```diff
@@ -76,6 +79,28 @@ func (s *StateSyncStore) insertStateSyncEvent(event *contractsapi.StateSyncedEve
 	})
 }
 
+// removeStateSyncEventsAndProofs removes state sync events and their proofs from the buckets in db
+func (s *StateSyncStore) removeStateSyncEventsAndProofs(stateSyncEventIDs []uint64) error {
+	return s.db.Update(func(tx *bolt.Tx) error {
+		eventsBucket := tx.Bucket(stateSyncEventsBucket)
+		proofsBucket := tx.Bucket(stateSyncProofsBucket)
+
+		for _, stateSyncEventID := range stateSyncEventIDs {
+			stateSyncEventIDKey := common.EncodeUint64ToBytes(stateSyncEventID)
+
+			if err := eventsBucket.Delete(stateSyncEventIDKey); err != nil {
+				s.logger.Info(fmt.Sprintf("failed to remove state sync event (ID=%d): %v", stateSyncEventID, err))
```

Comment:
I wouldn't log it here. Just return an error, and log the error in `state_sync_manager`.

Reply:
Dušan and I discussed this, we don't see state syncs and/or proofs removal as a critical operation so we wanted to proceed and execute the rest of the function (commitment insertion and proofs building). That's why an error is not propagated to a caller.

Reply:
I understand, but my point is to not place a logger in the store, rather have the store return an error, and just log it in the caller (`stateSyncManager`).

Reply:
Sorry, I made an oversight. Makes sense.

Reply:
If I returned the first error that was encountered, then other data (which might get removed during the next loop iteration) would not be removed.

Reply:
I don't think the retry mechanism will help in this situation. The db either has or hasn't this data under the given key. If some key is provided and the removal fails the first time, it will also fail during retry calls. There will just be N unnecessary calls which are also going to fail.

Reply:
Yes, but `Delete` doesn't fail if it doesn't have the key in the bucket it fails because tx is closed, it is not a write transaction, or a file error occurs. That then, requires a retry.

Reply:
Yes we are, but as I said, chances are slim, but `db` can still return an error (file is no longer available, `os` is returning some errors). I am fine if you don't want a retry mechanism here, deleting executing events is not a critical nor an important thing, they can exist in `db` forever, as far as I am concerned. I just wanted this to follow the same pattern we have in other components.

Reply:
returning errors instead of logging: 118901573a263bb979712ae7984a8c22d6246db5